examples/s/d/nn/_project/aggregate_surfaces.py

```diff
@@ -0,0 +1,236 @@
+# ======================================================================================
+
+
+def _aggregate(source_surfaces, operation):
+    """Aggregate a set of surfaces, return the result.
+
+    This is a very simplistic and minimalistic aggregation method meant to power this
+    example only. Do not use in production setting.
+    """
+
+    if operation == "mean":
+        return source_surfaces.apply(np.nanmean, axis=0)
+    if operation == "min":
+        return source_surfaces.apply(np.min, axis=0)
+    if operation == "max":
+        return source_surfaces.apply(np.max, axis=0)
+    if operation == "std":
+        return source_surfaces.apply(np.std, axis=0)
+
+    # In a real aggregation service, more options would of course be supported. However,
+    # in this example, we do not include anything beyond the basics.
+
+    raise NotImplementedError(
+        f"Aggregation method {operation} is not implemented in this example."
+    )
+
+
+def _parse_yaml(fname):
+    """Parse the yaml-file, return dict.
+
+    Args:
+        fname (Path): Absolute path to yaml file.
+    Returns:
+        dict
+    """
+
+    with open(fname, "r") as stream:
+        data = yaml.safe_load(stream)
+    return data
+
+
+def _metadata_filename(fname):
+    """From a regular filename, derive the corresponding metadata filename.
+
+    FMU standard: metadata filename = /path/.<stem.ext>.yml
+
+    """
+
+    return Path(fname.parent, "." + fname.name + ".yml")
+
+
+def _get_realization_ids(casepath):
+    """Given a path to a case on the disk, get the individual realizations."""
+
+    # In reality we would traverse the disk to find out, use Sumo search, or preferably
+    # ask ERT for it. For the sake of the example we just hardcode it here.
+
+    return [0, 1, 9]
+
+
+def _get_source_surfaces_from_disk(
+    casepath: Path, iter_name: str, realization_ids: list, relative_path: Path
+):
+    """Collect surfaces and metadata from disk.
+
+    This method collects the source surfaces from disk. Source surfaces are the
+    individual realization surfaces which shall be aggregated.
+
+    A similar method will exist for other sources than /scratch, e.g. Sumo.
+
+    Args:
+        casepath (Path): Absolute path to the case root.
+        iter_name (str): Name of the iteration (folder), e.g. "iter-0"
+        reals (list of ids): List of realization-ids, e.g. 0,1,2,3
+        relative_path (Path): Relative path below ERT RUNPATH to the surface
+
+
+    By combining the casepath and the relative path, parse the surfaces and metadata
+    and return surfaces as an xtgeo.Surfaces object and the metadata as a list of dicts.
+
+    """
+
+    collected_data = []
+
+    for real in realization_ids:
+        surfacepath = casepath / f"realization-{real}" / iter_name / relative_path
+        surface = xtgeo.surface_from_file(surfacepath)
+
+        metadatapath = _metadata_filename(surfacepath)
+        metadata = _parse_yaml(metadatapath)
+
```

Comment:
You can also use:
`metadata = fmu.dataio.read_metadata(surfacepath)   # this will get the assosiated metadata content`

Reply:
Using it now. Also added import to `__init__` to make the above syntax work.